astropy/cosmology/flrw.py

```diff
@@ -1485,6 +1454,99 @@ def arcsec_per_kpc_proper(self, z):
+        # subclassses of FlatFLRWMixin.
+
+        if other.__class__ is not self.__class__:
+            return NotImplemented
+
+        # check all parameters in other match those in 'self' and 'other' has
+        # no extra parameters (latter part should never happen b/c same class)
+        sps, ops = self.info("parameters"), other.info("parameters")
+        params_eq = (all((np.all(ops[k] == v) for k, v in sps.items()))
+                     and not (sps.keys() ^ ops.keys()))
+
+        return params_eq or NotImplemented
+
+
+class FlatFLRWMixin(FlatCosmologyMixin):
+    """
+    Mixin class for flat FLRW cosmologies. Do NOT instantiate directly.
+    Must precede the base class in the multiple-inheritance so that this mixin's
+    ``__init__`` proceeds the base class'.
+    Note that all instances of ``FlatFLRWMixin`` are flat, but not all
+    flat cosmologies are instances of ``FlatFLRWMixin``. As example,
+    ``LambdaCDM`` **may** be flat (for the a specific set of parameter values),
+    but ``FlatLambdaCDM`` **will** be flat.
+    """
+
+    def __init__(self, *args, **kw):
+        super().__init__(*args, **kw)
+        # Do some twiddling after the fact to get flatness
+        self._Ode0 = 1.0 - self._Om0 - self._Ogamma0 - self._Onu0
+        self._Ok0 = 0.0
+
+    @property  # no longer a Parameter
+    def Ode0(self):
+        """Omega dark energy; dark energy density/critical density at z=0"""
+        return self._Ode0
+
+    def __equiv__(self, other):
+        """flat-FLRW equivalence. Use ``.is_equivalent()`` for actual check!
+
+        Parameters
+        ----------
+        other : `~astropy.cosmology.FLRW` subclass instance
+            The object in which to compare.
+
+        Returns
+        -------
+        `True` or `NotImplemented`
+            `True` if 'other' is of the same class / non-flat class (e.g.
+            ``FlatLambdaCDM`` and ``LambdaCDM``) has matching parameters
+            and parameter values. `NotImplemented` otherwise.
+
+        """
+        # check if case (1): same class & parameters
+        if isinstance(other, FlatFLRWMixin):
+            return super().__equiv__(other)
+
+        # check cases (3, 4), if other is the non-flat version of this class
+        # mro: [0: flat-class, 1: FlatFLRWMixin, 2: FlatCosmologyMixin, 3: non-flat]
+        elif other.__class__ is not self.__class__.mro()[3]:
+            return NotImplemented
+
+        # check if have equivalent parameters
+        # check all parameters in other match those in 'self' and 'other' has
+        # no extra parameters (case (2)) except for 'Ode0' and that other
+        sps, ops = self.info("parameters"), other.info("parameters")
+        params_eq = (
+            all((np.all(ops[k] == v) for k, v in sps.items()))  # params equal
+            and not ((sps.keys() ^ ops.keys()) - {"Ode0"})  # no extra params
+            # flatness conditions
+            and other.Ok0 == 0.0
```

Comment:
If we were to let `other` do this, it would have to be written differently, as `all((np.all(getattr(ops, k, None) == v) for k, v in sps.items()))`. Indeed, perhaps one can more generally write this as
```
all(np.all(getattr(self, parameter) == getattr(other, parameter, None)) for parameter in self.parameters)
```
Might even be a bit faster...

Reply:
I think the key-check is important.
``getattr(self, parameter)`` can be `None` and if "parameter" doesn't exist on ``other``, than it should raise an error.